### ISSUE TITLE
[types] documenting features as typescript

### ISSUE THREAD

**Issue by sonicolasj** [zloirock/core-js#1064]
Hi!

Context: in the README's features, there are block codes used to shown what's included in a feature (ex: ). These blocks look a lot like typescript type declarations, but aren't.

First: shouldn't these blocks be tagged as typescript instead of javascript (sed /\`\`\`js/\`\`\`ts/g), so we can have proper syntax coloration?

Second: as a stepping point to #568, we could make these blocks correct typescript, so TS users can copy/paste them safely in their codebase when importing the polyfills (especially proposals).

If that's interesting, I could probably start a PR on that!

**Comment by zloirock** [comment]
Hi. This is not a TypeScript. TypeScript is not the only superset of JavaScript. JavaScript haven't own IDL for definition typing signatures, so this a kind of pseudocode.

If you wanna help to add TypeScript definitions - [it's another issue](https://github.com/zloirock/core-js/issues/568) - feel free to work on it.

When JavaScript will have own IDL (for example, [this proposal](https://github.com/tc39/proposal-idl)) or at least ["types as comments"](https://github.com/giltayar/proposal-types-as-comments/) - `core-js` will be adapted to it.

**Comment by sonicolasj** [comment]
> JavaScript haven't own IDL for definition typing signatures, so this a kind of pseudocode.

Yes, of course!
But this pseudo-code already looks like a lot like TypeScript. In my opinion, writing them would not really change anything for JavaScript users, and help TypeScript users by giving them a snippet to copy/paste while waiting for the proper types definition to be available!

Ex: current js block

```js
class Set {
  constructor(iterable?: Iterable<value>): Set;
  add(key: any): this;
  clear(): void;
  delete(key: any): boolean;
  forEach((value: any, key: any, target: any) => void, thisArg: any): void;
  has(key: any): boolean;
  values(): Iterator<value>;
  keys(): Iterator<value>;
  entries(): Iterator<[value, value]>;
  @@iterator(): Iterator<value>;
  readonly attribute size: number;
}
```

ts version:

```typescript
class Set<TValue> {
    constructor(iterable?: Iterable<TValue>)
    add(key: TValue): this;
    clear(): void;
    delete(key: TValue): boolean;
    forEach(fn: (value: TValue, key: number, target: this) => void, thisArg?: any): void;
    has(key: TValue): boolean;
    values(): Iterator<TValue>;
    keys(): Iterator<TValue>;
    entries(): Iterator<[number, TValue]>;
    [Symbol.iterator](): Iterator<TValue>;
    readonly size: number;
}
```

> If you wanna help to add TypeScript definitions - https://github.com/zloirock/core-js/issues/568 - feel free to work on it.

Yes, I'm planning to write some typings, as I currently did in my code!

**Comment by zloirock** [comment]
> `entries(): Iterator<[number, TValue]>;`

Not exactly.

> But this pseudo-code already looks like a lot like TypeScript. In my opinion, writing them would not really change anything for JavaScript users, and help TypeScript users by giving them a snippet to copy/paste while waiting for the proper types definition to be available!

Hmm, why not... Ok, feel free to do it.

**Comment by sonicolasj** [comment]
>> entries(): Iterator<[number, TValue]>;

> Not exactly.

Good catch, it was me assuming it worked like that. Same goes for `forEach` then!

**Comment by zloirock** [comment]
@sonicolasj hey, any progress?

**Comment by sonicolasj** [comment]
Hi!
Yes, I'm documenting everything when using stuff (for the proposals mainly), I'll push when I have something good to show!

**Comment by zloirock** [comment]
Due to the lack of activity here, I'm adding a `help wanted` label.

**Comment by zloirock** [comment]
https://github.com/Kingwl/core-dts
